manim/config.py

```diff
@@ -1,326 +1,429 @@
-        video_group = parser.add_mutually_exclusive_group()
-        video_group.add_argument(
-            "--video_dir",
-            help="Directory to write file tree for video",
-        )
-        parser.add_argument(
-            "--tex_dir",
-            help="Directory to write tex",
-        )
-        parser.add_argument(
-            "--text_dir",
-            help="Directory to write text",
-        )
-        parser.add_argument(
-            "--tex_template",
-            help="Specify a custom TeX template file",
+                 "it should be different from the scene class name",
         )
-        return parser.parse_args()
-
-    except argparse.ArgumentError as err:
-        logger.error(str(err))
-        sys.exit(2)
-
-
-def get_configuration(args):
-    file_writer_config = {
-        # By default, write to file
-        "write_to_movie": (args.write_to_movie or not args.save_last_frame) and not args.dry_run,
-        "save_last_frame": args.save_last_frame and not args.dry_run,
-        "save_pngs": args.save_pngs,
-        "save_as_gif": args.save_as_gif,
-        # If -t is passed in (for transparent), this will be RGBA
-        "png_mode": "RGBA" if args.transparent else "RGB",
-        "movie_file_extension": ".mov" if args.transparent else ".mp4",
-        "file_name": args.file_name,
-        "input_file_path": args.file,
-    }
-    config = {
-        "file": args.file,
-        "scene_names": args.scene_names,
-        "open_video_upon_completion": args.preview,
-        "show_file_in_finder": args.show_file_in_finder,
-        "file_writer_config": file_writer_config,
-        "quiet": args.quiet or args.write_all,
-        "ignore_waits": args.preview,
-        "write_all": args.write_all,
-        "start_at_animation_number": args.start_at_animation_number,
-        "end_at_animation_number": None,
-        "sound": args.sound,
-        "leave_progress_bars": args.leave_progress_bars,
-        "media_dir": args.media_dir,
-        "video_dir": args.video_dir,
-        "tex_dir": args.tex_dir,
-        "text_dir": args.text_dir,
-        "tex_template": args.tex_template,
-    }
-
-    # Camera configuration
-    config["camera_config"] = get_camera_configuration(args)
-
-    # Arguments related to skipping
-    stan = config["start_at_animation_number"]
-    if stan is not None:
-        if "," in stan:
-            start, end = stan.split(",")
-            config["start_at_animation_number"] = int(start)
-            config["end_at_animation_number"] = int(end)
-        else:
-            config["start_at_animation_number"] = int(stan)
 
-    config["skip_animations"] = any([
-        file_writer_config["save_last_frame"],
-        config["start_at_animation_number"],
-    ])
-    return config
+    # Note the following use (action='store_const', const=True),
+    # instead of using the built-in (action='store_true').  The reason
+    # is that these two are not equivalent.  The latter is equivalent
```

Comment:
nit: it's redundant to say the two aren't equivalent, you can just say why you chose the first version.

Reply:
Always happy to be nitpicked about comments/documentation. Fixed in latest commit.